### PR TITLE
[CROSSDATA-416] [DRIVER] Removing ask pattern

### PR DESCRIPTION
**Pull request by miguel0afd**
This PR removes the ask pattern from the Proxy Actor using a map of Promises stored by request id.

### CODE REVIEW

**Comment by jjlopezm** [comment]
+1

File: driver/src/main/scala/com/stratio/crossdata/driver/Driver.scala [hunk 1]
```diff
@@ -97,27 +150,36 @@ class Driver private(driverConf: DriverConf,
...
-      case Pattern(add,jar,path)=>addJar(path.trim)
-      case _=>
+      case Pattern(add,jar,path) => addJar(path.trim)
+      case _ =>
         val sqlCommand = new SQLCommand(query, retrieveColNames = driverConf.getFlattenTables)
         val secureCommand = CommandEnvelope(sqlCommand, driverSession)
-        new SQLResponse(sqlCommand.requestId, askCommand[SQLReply](proxyActor, secureCommand).map(_.sqlResult)) {
+        val futureReply = askCommand(proxyActor, secureCommand).map{
+          case SQLReply(_, sqlResult) => sqlResult
+          case reply => throw new RuntimeException(s"SQLResult expected. Received: $reply")
```
Comment (darroyocazorla) [inline driver/src/main/scala/com/stratio/crossdata/driver/Driver.scala]: SQLReply?

File: driver/src/main/scala/com/stratio/crossdata/driver/Driver.scala [hunk 2]
```diff
@@ -97,27 +150,36 @@ class Driver private(driverConf: DriverConf,
...
     * @param path The path of the JAR
     * @return A SQLResponse with the id and the result set.
     */
   def addJar(path: String): SQLResponse = {
     val addJarCommand = new AddJARCommand(path)
     val secureCommand = CommandEnvelope(addJarCommand, driverSession)
-    new SQLResponse(addJarCommand.requestId, askCommand[SQLReply](proxyActor, secureCommand).map(_.sqlResult))
+    val futureReply = askCommand(proxyActor, secureCommand).map{
+      case SQLReply(_, sqlResult) => sqlResult
+      case reply => throw new RuntimeException(s"SQLResult expected. Received: $reply")
```
Comment (darroyocazorla) [inline driver/src/main/scala/com/stratio/crossdata/driver/Driver.scala]: Same as above

File: driver/src/main/scala/com/stratio/crossdata/driver/Driver.scala [hunk 3]
```diff
@@ -166,8 +228,12 @@ class Driver private(driverConf: DriverConf,
 
   // TODO remove infinite duration
```
Comment (darroyocazorla) [inline driver/src/main/scala/com/stratio/crossdata/driver/Driver.scala]: TODOs can be removed

File: driver/src/main/scala/com/stratio/crossdata/driver/Driver.scala [hunk 4]
```diff
@@ -166,8 +228,12 @@ class Driver private(driverConf: DriverConf,
...
   // TODO remove askPattern
-  private def askCommand[T <: ServerReply : ClassTag](remoteActor: ActorRef, message: Any, timeout: FiniteDuration = 1 day): Future[T] = {
-    remoteActor.ask(message)(timeout).mapTo[T]
+  private def askCommand(remoteActor: ActorRef,
+                         message: CommandEnvelope,
```
Comment (darroyocazorla) [inline driver/src/main/scala/com/stratio/crossdata/driver/Driver.scala]: Maybe:
private def askCommand(command: CommandEnvelope)(implicit val remoteActor: ActorRef) and replace val proxy actor with implicit val proxyActor or even use it directly.

File: driver/src/main/scala/com/stratio/crossdata/driver/actor/ProxyActor.scala [hunk 5]
```diff
@@ -40,29 +47,75 @@ class ProxyActor(clusterClientActor: ActorRef, driver: Driver) extends Actor {
...
+    case secureSQLCommand @ CommandEnvelope(sqlCommand: SQLCommand, _) =>
+      logger.info(s"Sending query: ${sqlCommand.sql} with requestID=${sqlCommand.requestId} & queryID=${sqlCommand.queryId}")
+      clusterClientActor ! ClusterClient.Send(ProxyActor.ServerPath, secureSQLCommand, localAffinity = false)
 
     /* TODO: This is a dirty trick to keep temporary tables synchronized at each XDContext
         it should be fixed as soon as Spark version is updated to 1.6 since it'll enable.
         WARNING: This disables creation cancellation commands and exposes the system behaviour to client-side code.
      */
-    case secureSQLCommand @ CommandEnvelope(sqlCommand @ SQLCommand(sql @ catalogOpExp(), _, _, _, _), _) =>
+    case secureSQLCommand @ CommandEnvelope(sqlCommand @ SQLCommand(sql @ catalogOpExp(), _, _, _), _) =>
```
Comment (darroyocazorla) [inline driver/src/main/scala/com/stratio/crossdata/driver/actor/ProxyActor.scala]: Dead code! It should be the first case

**Pull request merged**